app/pages/patient/patientinfo.js

```diff
@@ -399,6 +423,22 @@ var PatientInfo = React.createClass({
       return 'Diagnosis date needs to be a valid date';
     }
 
+    currentDateObj = currentDateObj || new Date();
+    var birthdayDateObj = sundial.parseFromFormat(birthday, FORM_DATE_FORMAT);
+    var diagnosisDateObj = sundial.parseFromFormat(diagnosisDate, FORM_DATE_FORMAT);
+
+    if (birthdayDateObj > currentDateObj) {
+      return 'Date of birth cannot be in the future!';
+    }
+
+    if (diagnosisDateObj > currentDateObj) {
+      return 'Diagnosis date cannot be in the future!';
+    }
+
+    if (birthdayDateObj > diagnosisDateObj) {
+      return 'Diagnosis cannot be before date of birth!';
+    }
```

Comment:
Yay, thanks for adding these logic checks.

(Note to self: Trello card created [here](https://trello.com/c/LeNQuuJd) for tracking PR and eventually adding testing instructions for QA of the small functionality changes introduced here.)